scipy/special/_basic.pyi

```diff
@@ -0,0 +1,271 @@
+
+digamma = psi
+
+def diric(x: NDArray, n: DecimalNumber) -> NDArray: ...
+def jnjnp_zeros(nt: int) -> _ZerosOrder: ...
+def jnyn_zeros(n: int, nt: int) -> _ZerosOrder: ...
+def jn_zeros(n: int, nt: int) -> NDArray: ...
+def jnp_zeros(n: int, nt: int) -> NDArray: ...
+def yn_zeros(n: int, nt: int) -> NDArray: ...
+def ynp_zeros(n: int, nt: int) -> NDArray: ...
+def y0_zeros(nt: int, complex: Literal[False] = ...) -> _PointsDerivate: ...
+def y1_zeros(nt: int, complex: Literal[False] = ...) -> _PointsDerivate: ...
+def y1p_zeros(nt: int, complex: Literal[False] = ...) -> _PointsDerivate: ...
+@overload
+def jvp(v: NDArray, z: _ComplexNumber | NDArray, n: Literal[1] = ...) -> NDArray: ...
+@overload
+def jvp(v: NDArray, z: _ComplexNumber | NDArray, n: int) -> NDArray: ...
+@overload
+def jvp(v: DecimalNumber | NDArray, z: NDArray, n: Literal[1] = ...) -> NDArray: ...
+@overload
+def jvp(v: DecimalNumber | NDArray, z: NDArray, n: int) -> NDArray: ...
+@overload
+def jvp(v: DecimalNumber, z: _ComplexNumber, n: Literal[1] = ...) -> DecimalNumber: ...
+@overload
+def jvp(v: DecimalNumber, z: _ComplexNumber, n: int) -> DecimalNumber: ...
+@overload
+def yvp(v: NDArray, z: DecimalNumber | NDArray, n: int = ...) -> NDArray: ...
+@overload
+def yvp(v: DecimalNumber | NDArray, z: NDArray, n: int = ...) -> NDArray: ...
+@overload
+def yvp(v: DecimalNumber, z: DecimalNumber, n: int = ...) -> DecimalNumber: ...
+@overload
+def kvp(v: DecimalNumber, z: _ComplexNumber, n: int = 1) -> DecimalNumber: ...
+@overload
+def kvp(v: NDArray, z: NDArray | _ComplexNumber, n: int = 1) -> NDArray: ...
+@overload
+def kvp(v: NDArray | DecimalNumber, z: NDArray, n: int = 1) -> NDArray: ...
+@overload
+def ivp(v: NDArray, z: DecimalNumber | NDArray, n: int = ...) -> NDArray: ...
+@overload
+def ivp(v: DecimalNumber | NDArray, z: NDArray, n: int = ...) -> NDArray: ...
+@overload
+def ivp(v: DecimalNumber, z: DecimalNumber, n: int = ...) -> DecimalNumber: ...
+@overload
+def h1vp(v: NDArray, z: DecimalNumber | NDArray, n: int = ...) -> NDArray: ...
+@overload
+def h1vp(v: DecimalNumber | NDArray, z: NDArray, n: int = ...) -> NDArray: ...
+@overload
+def h1vp(v: DecimalNumber, z: DecimalNumber, n: int = ...) -> DecimalNumber: ...
+@overload
+def h2vp(v: NDArray, z: DecimalNumber | NDArray, n: int = ...) -> NDArray: ...
+@overload
+def h2vp(v: DecimalNumber | NDArray, z: NDArray, n: int = ...) -> NDArray: ...
+@overload
+def h2vp(v: DecimalNumber, z: DecimalNumber, n: int = ...) -> DecimalNumber: ...
+def riccati_jn(n: int, x: float) -> _PointsDerivate: ...
+def riccati_yn(n: int, x: float) -> _PointsDerivate: ...
+def erf_zeros(nt: int) -> NDArray: ...
+def fresnelc_zeros(nt: int) -> NDArray: ...
+def fresnels_zeros(nt: int) -> NDArray: ...
+def fresnel_zeros(nt: int) -> _PointsDerivate: ...
+@overload
+def assoc_laguerre(x: NDArray, n: int, k: float = ...) -> NDArray: ...
+@overload
+def assoc_laguerre(x: DecimalNumber, n: int, k: float = ...) -> DecimalNumber: ...
```

Comment:
The return type seems to always be an `np.float64` scalar. It shouldn't be a union (which `DecimalNumber` is).

Reply:
Please double check for the other `-> DecimalNumber`'s as well.